Editor/CoreAPI/GameLiftFleetManager.cs

```diff
@@ -35,36 +35,32 @@ public GameLiftFleetManager(IAmazonGameLiftWrapper amazonGameLiftWrapper)
 
         public async Task<CreateAnywhereFleetResponse> CreateFleet(string fleetName)
         {
-            if (_amazonGameLiftWrapper != null)
+            if (_amazonGameLiftWrapper == null)
             {
-                var success = await CreateCustomLocationIfNotExists(FleetLocation);
-                if (success)
-                {
-                    var fleetId = await CreateFleet(ComputeType.ANYWHERE, FleetLocation, fleetName);
-                    if (fleetId == null)
-                    {
-                        return Response.Fail(new CreateAnywhereFleetResponse
-                        {
-                            ErrorCode = ErrorCode.InvalidFleetName
-                        });
-                    }
+                return Response.Fail(new CreateAnywhereFleetResponse { ErrorCode = ErrorCode.AccountProfileMissing });
+            }
 
-                    return Response.Ok(new CreateAnywhereFleetResponse()
-                    {
-                        FleetId = fleetId,
-                        FleetName = fleetName
-                    });
-                }
+            if (string.IsNullOrWhiteSpace(fleetName))
+            {
+                return Response.Fail(new CreateAnywhereFleetResponse { ErrorCode = ErrorCode.InvalidFleetName });
+            }
 
-                return Response.Fail(new CreateAnywhereFleetResponse
-                {
-                    ErrorCode = ErrorCode.CustomLocationCreationFailed
-                });
+            var createLocationSuccess = await CreateCustomLocationIfNotExists(FleetLocation);
+            if (!createLocationSuccess)
+            {
+                return Response.Fail(new CreateAnywhereFleetResponse { ErrorCode = ErrorCode.CustomLocationCreationFailed });
+            }
+
+            var fleetId = await CreateFleet(ComputeType.ANYWHERE, FleetLocation, fleetName);
+            if (string.IsNullOrWhiteSpace(fleetId))
+            {
+                return Response.Fail(new CreateAnywhereFleetResponse { ErrorCode = ErrorCode.CreateFleetFailed });
```

Comment:
Ideally, if this happens we add the exception message to error box. I think this was a case that was missed in the error box PR, so I've added an item to the bug tracker